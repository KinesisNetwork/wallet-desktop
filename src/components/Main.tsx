import * as React from 'react'

import { ConnectionSettings } from '@components/ConnectionSettings'
import { Payee } from '@components/Payee'
import { Sidebar } from '@components/Sidebar'
import { AccountPage } from '@containers/AccountPage'
import { CreateWallet } from '@containers/CreateWallet'
import { WalletView } from '@types'

export interface Props {
  activeView: WalletView
}

export class Main extends React.PureComponent<Props> {
  displayName = 'Main'

  viewMap = (view: WalletView) => {
    switch (view) {
      case WalletView.create:
        return <CreateWallet />
      case WalletView.dashboard:
        return <AccountPage />
      case WalletView.settings:
        return <ConnectionSettings />
      case WalletView.payees:
        return <Payee />
      default:
        return <div />
    }
  }

  render() {
    return (
      <div className="columns is-marginless" style={{ height: '100vh' }}>
        <div className="column is-one-quarter" style={{ backgroundColor: '#2b3e50' }}>
          <Sidebar />
        </div>
        <div className="column">{this.viewMap(this.props.activeView)}</div>
      </div>
    )
  }
}
