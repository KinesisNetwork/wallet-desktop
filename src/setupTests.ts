import { configure } from 'enzyme'
import * as Adapter from 'enzyme-adapter-react-16'

export default configure({ adapter: new Adapter() })
