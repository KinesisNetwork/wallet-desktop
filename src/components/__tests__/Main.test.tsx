import { shallow } from 'enzyme';
import * as React from 'react'

import { Main } from '@components/Main'
import '../../setupTests'

describe('Main', () => {
  it('renders correctly', () => {
    const wrapper = shallow(<Main activeView={0} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('always renders Sidebar plus the relevant component', () => {
    const createWalletWrapper = shallow(<Main activeView={0} />)
    const connectionSettingsWrapper = shallow(<Main activeView={1} />)
    const accountPageWrapper = shallow(<Main activeView={2} />)
    const payeeWrapper = shallow(<Main activeView={3} />)

    expect(createWalletWrapper.find('Sidebar')).toHaveLength(1)
    expect(createWalletWrapper.find('Connect(CreateWallet)')).toHaveLength(1)
    expect(connectionSettingsWrapper.find('Sidebar')).toHaveLength(1)
    expect(connectionSettingsWrapper.find('ConnectionSettings')).toHaveLength(1)
    expect(accountPageWrapper.find('Sidebar')).toHaveLength(1)
    expect(accountPageWrapper.find('Connect(AccountPage)')).toHaveLength(1)
    expect(payeeWrapper.find('Sidebar')).toHaveLength(1)
    expect(payeeWrapper.find('Payee')).toHaveLength(1)
  })
})
