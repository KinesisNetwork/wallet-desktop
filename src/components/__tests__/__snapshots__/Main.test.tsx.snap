// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main renders correctly 1`] = `
<div
  className="columns is-marginless"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="column is-one-quarter"
    style={
      Object {
        "backgroundColor": "#2b3e50",
      }
    }
  >
    <Sidebar />
  </div>
  <div
    className="column"
  >
    <Connect(CreateWallet) />
  </div>
</div>
`;

exports[`Main renders correctly 2`] = `
<div
  className="columns is-marginless"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="column is-one-quarter"
    style={
      Object {
        "backgroundColor": "#2b3e50",
      }
    }
  >
    <Sidebar />
  </div>
  <div
    className="column"
  >
    <ConnectionSettings />
  </div>
</div>
`;

exports[`Main renders correctly 3`] = `
<div
  className="columns is-marginless"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="column is-one-quarter"
    style={
      Object {
        "backgroundColor": "#2b3e50",
      }
    }
  >
    <Sidebar />
  </div>
  <div
    className="column"
  >
    <Connect(AccountPage) />
  </div>
</div>
`;

exports[`Main renders correctly 4`] = `
<div
  className="columns is-marginless"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="column is-one-quarter"
    style={
      Object {
        "backgroundColor": "#2b3e50",
      }
    }
  >
    <Sidebar />
  </div>
  <div
    className="column"
  >
    <Payee />
  </div>
</div>
`;
